pkg/util/winutil/pdhutil/pdhcounter.go

```diff
@@ -108,20 +110,38 @@ func (counter *pdhCounter) ShouldInit() bool {
 		// already initialized
 		return false
 	}
+	var initFailLimit = config.Datadog.GetInt("windows_counter_init_failure_limit")
+	if initFailLimit > 0 && counter.initFailCount >= initFailLimit {
+		counter.initError = fmt.Errorf("Counter exceeded the maximum number of failed initialization attempts. This error indicates that the Windows performance counter database may need to be rebuilt.")
+		// attempts exceeded
+		return false
+	}
 	return true
 }
 
+func (counter *pdhCounter) countInitError(err error) error {
+	counter.initFailCount += 1
```

Comment:
this function is not called once the fail limit is reached.

I thought it could still be helpful to log the number of failures even when the limit is 0/disabled.
https://github.com/DataDog/datadog-agent/pull/14642/files#diff-1f19feefb3db6abaf147acef2e895624d71a9ee9bfe8b49762ba9089d3e9b718R130

Reply:
Then first "if" condition, theoretically is a moot point?

Reply:
the function is not called once the fail limit is reached. But once called it can increment the count TO the limit, in which case the first `if` condition will be true.